resource.cpp

```diff
@@ -1957,3 +1957,18 @@ bool isValidRConfigGroup(const QString &str)
 
     return result == groupList.size();
 }
+
+/*! Helper function to expose \p resource item as a different type through REST API than internally defined. */
+QVariant R_ItemToRestApiVariant(const ResourceItem *item)
+{
+    if (item)
+    {
+        const ResourceItemDescriptor rid = item->descriptor();
```

Comment:
Better use a reference here, otherwise the descriptor is copied each time.

`const ResourceItemDescriptor &rid = item->descriptor();`

Reply:
Ah right, updated :+1: